wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/registry/ServerCapabilitiesOverlay.java

```diff
@@ -113,11 +116,13 @@ public SignatureHelpOptions getSignatureHelpProvider() {
         return result;
     }
 
-    public TextDocumentSyncKind getTextDocumentSync() {
+    public Either<TextDocumentSyncKind, TextDocumentSyncOptions> getTextDocumentSync() {
```

Comment:
Do we really need to merge the whole thing, or can we just return TextDocumentSyncKind here? I don't think we're doing anything on the client side with it.